api/src/test/java/edu/wpi/first/shuffleboard/api/sources/recording/SerializationTest.java

```diff
@@ -131,6 +132,20 @@ public void testUpdateFileWithNewDataTypes() throws IOException {
     assertEquals(data, last.getData(), "Second update was wrong");
   }
 
+  @Test
+  public void testEncodeRecodeWithMarkers() throws IOException {
+    final Path file = Files.createTempFile("testEncodeRecodeWithMarkers", ".sbr");
```

Comment:
Do you ever clean these files up? Are you certain you aren't leaking temporary files all over your file system doing this?

You should really use Junit Pioneer's temporary directory extension:
https://junit-pioneer.org/docs/temp-directory/